src/core/github-api.js

```diff
@@ -20,32 +20,28 @@ export function githubRequestHeaders(conf) {
   return headers;
 }
 
-export async function getRateLimit(conf) {
-  const headers = githubRequestHeaders(conf);
-  const request = new Request(`https://api.github.com/rate_limit`, {
-    mode: "cors",
-    referrerPolicy: "no-referrer",
-    headers,
-  });
-  const responseJSON = await fetch(request).then(r => r.json());
-  return responseJSON.rate.remaining;
+export function checkLimitReached(response) {
+  if (response.ok) return false;
+  const { headers, status } = response;
+  if (status === 403 && headers.get("X-RateLimit-Remaining") === "0") {
+    if (typeof checkLimitReached.warned === "undefined") {
+      checkLimitReached.warned = true;
+      const msg = `You have run out of github requests. Some github assets will not show up.`;
+      pub("warning", msg);
+      window.pub = pub;
+    }
+    return true;
+  }
+  // Not sure we can reach here.
```

Comment:
```suggestion
  return true;
```

Reply:
if the response code is neither 200 nor 403. 
(Means if we neither get the correct response, nor are we out of requests) the function would in this case return true. 
Is this okay?

Reply:
I think I should add a warning for such cases (when !response.ok) so the viewer knows something is not right.

Reply:
```javascript
if (response.ok) return false;
else pub("warning", "unexpected response received from github");
  const { headers, status } = response;
  if (status === 403 && headers.get("X-RateLimit-Remaining") === "0") {
    if (typeof checkLimitReached.warned === "undefined") {
      checkLimitReached.warned = true;
      const msg = `You have run out of github requests. Some github assets will not show up.`;
      pub("warning", msg);
      window.pub = pub;
    }
   return true;
  }
return false;
```
This feels like the correct thing to do

Reply:
Ah, right, ok... yeah, would be good to warn and show the HTTP error code.

Reply:
> pub("warning", "unexpected response received from github");

Yep, see other HTTP errors we emit. Can just copy those.

Reply:
Don't need to warn. It's being done by both core/contrib (for contributors) and core/github-api for issues and comments.

Reply:
Excellent, thanks for checking!